api/tests/unit/mappers/test_unit_mappers_engine.py

```diff
@@ -0,0 +1,463 @@
+    expected_result = IntegrationModel(base_url=base_url, api_key=api_key)
+
+    # When
+    result = engine.map_integration_to_engine(integration)
+
+    # Then
+    assert result == expected_result
+
+
+def test_map_integration_to_engine__none__return_expected() -> None:
+    # When
+    result = engine.map_integration_to_engine(None)
+
+    # Then
+    assert result is None
+
+
+def test_map_integration_to_engine__dynatrace__return_expected() -> None:
+    # Given
+    base_url = "http://someurl"
+    api_key = "test"
+    entity_selector = "*"
+
+    integration = DynatraceConfiguration(
+        base_url=base_url,
+        api_key=api_key,
+        entity_selector=entity_selector,
+    )
+    expected_result = IntegrationModel(
+        base_url=base_url,
+        api_key=api_key,
+        entity_selector=entity_selector,
+    )
+
+    # When
+    result = engine.map_integration_to_engine(integration)
+
+    # Then
+    assert result == expected_result
+
+
+def test_map_webhook_config_to_engine__return_expected() -> None:
+    # Given
+    url = "http://someurl"
+    secret = "test"
+
+    webhook_config = WebhookConfiguration(
+        url=url,
+        secret=secret,
+    )
+    expected_result = WebhookModel(
+        url=url,
+        secret=secret,
+    )
+
+    # When
+    result = engine.map_webhook_config_to_engine(webhook_config)
+
+    # Then
+    assert result == expected_result
+
+
+def test_map_webhook_config_to_engine__none__return_expected() -> None:
+    # When
+    result = engine.map_webhook_config_to_engine(None)
+
+    # Then
+    assert result is None
+
+
+def test_map_feature_state_to_engine__return_expected(
+    feature: "Feature",
+    feature_state: FeatureState,
+) -> None:
+    # Given
+    expected_result = FeatureStateModel(
+        feature=FeatureModel(
+            id=feature.id,
+            name="Test Feature1",
+            type="STANDARD",
+        ),
+        enabled=False,
+        django_id=feature_state.id,
+        feature_segment=None,
+        featurestate_uuid=feature_state.uuid,
+        feature_state_value=None,
+        multivariate_feature_state_values=[],
+    )
+
+    # When
+    result = engine.map_feature_state_to_engine(
+        feature_state,
+        [],
+    )
+
+    # Then
+    assert result == expected_result
+
+
+def test_map_feature_state_to_engine__segment_multivariate__return_expected(
+    segment_multivariate_feature_state: FeatureState,
+    multivariate_feature: "Feature",
+) -> None:
+    # Given
+    mv_fs_values = list(
+        segment_multivariate_feature_state.multivariate_feature_state_values.all()
+    )
+    expected_result = FeatureStateModel(
+        feature=FeatureModel(
+            id=multivariate_feature.id,
+            name="feature",
+            type="MULTIVARIATE",
+        ),
+        enabled=False,
+        django_id=segment_multivariate_feature_state.id,
+        feature_segment=None,
```

Comment:
As per my comment above. I think there's something slightly off with this test. We seem to be trying to test some segment related behaviour (as well as MV?) but are then asserting that we get None for the feature segment.

Reply:
Thanks! Fixed (and test name modified to better reflect the tested behaviour).